docs/src/architecture/accounts.md

```diff
@@ -8,44 +8,51 @@ An account is an entity which holds assets and defines rules of how these assets
 In the above:
 
 * **Account ID** is a unique identifier (8 bytes) of an account which does not change throughout its lifetime. It also specifies the type of the underlying account. 
-* **Storage** is user-defined data which can be stored in an account. The data is stored as key-value map which maps 256-bit keys to 256-bit values.
-* **Nonce** is the account's nonce increments every time the account changes its state. 
+* **Storage** is user-defined data which can be stored in an account. The data can be stored in a simple `SparseMerkleTree` (256 Word slots), or if additional capacity is needed in a `MerkleStore` allowing the user to store any Merkle structures users need.
+* **Nonce** is a counter that increments every time the account changes its state. 
```

Comment:
Should we mention here that the counter needs to be monotonically increasing, but it doesn't need to be incremented by exactly 1? E.g., we could increment the counter by 2, 3 etc. anything that is less than $2^{32}$.